src/snakemake/logging.py

```
@@ -738,24 +738,44 @@ def setup_logfile(self, workdir: Optional[os.PathLike] = None) -> None:
-        # Remove and close all handlers - this should mostly clean up the global logger instance.
+        # Flush, remove and close all handlers - this should mostly clean up the global logger instance.
         for handler in list(self.logger.handlers):
-            self.logger.removeHandler(handler)
-            handler.close()
+            try:
+                handler.acquire()
+
+                try:
+                    handler.flush()
+                except Exception:
+                    pass
```

Ruff (0.15.13)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)